src/main/kotlin/no/nav/familie/ef/iverksett/brev/JournalførVedtaksbrevTask.kt

```diff
@@ -41,21 +44,54 @@ class JournalførVedtaksbrevTask(private val iverksettingRepository: Iverksettin
                                 dokumenttype = Dokumenttype.VEDTAKSBREV_OVERGANGSSTØNAD,
                                 tittel = lagDokumentTittel(iverksett.fagsak.stønadstype, iverksett.vedtak.vedtaksresultat))
 
-        val journalpostId = journalpostClient.arkiverDokument(
-                ArkiverDokumentRequest(
-                        fnr = iverksett.søker.personIdent,
-                        forsøkFerdigstill = true,
-                        hoveddokumentvarianter = listOf(dokument),
-                        fagsakId = iverksett.fagsak.eksternId.toString(),
-                        journalførendeEnhet = iverksett.søker.tilhørendeEnhet,
-                        eksternReferanseId = "$behandlingId-vedtaksbrev"
-                ),
-                iverksett.vedtak.beslutterId
-        ).journalpostId
 
-        tilstandRepository.oppdaterJournalpostResultat(behandlingId = behandlingId,
-                                                       JournalpostResultat(journalpostId = journalpostId)
+        val arkiverDokumentRequest = ArkiverDokumentRequest(
+            fnr = iverksett.søker.personIdent,
+            forsøkFerdigstill = true,
+            hoveddokumentvarianter = listOf(dokument),
+            fagsakId = iverksett.fagsak.eksternId.toString(),
+            journalførendeEnhet = iverksett.søker.tilhørendeEnhet,
+            eksternReferanseId = "$behandlingId-vedtaksbrev"
         )
+
+        if(iverksett.vedtak.brevmottakere != null){
+            val journalførteIdenter: List<String> =
+                    tilstandRepository.hentJournalpostResultatBrevmottakere(behandlingId)?.keys?.toList() ?: emptyList()
+
+            iverksett.vedtak.brevmottakere.mottakere.map {
+                if (!journalførteIdenter.contains(it.ident)) {
+                    val journalpostId = journalpostClient.arkiverDokument(
+                            arkiverDokumentRequest.copy(
```

Comment:
Tror vi må endre `eksternReferanseId` for hver brevmottaker også - den skal nok være unik mellom hver request.